components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/GatewayPolicyDeployer.java

```diff
@@ -0,0 +1,191 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.apimgt.gateway;
+
+import com.google.gson.Gson;
+import org.apache.axis2.AxisFault;
+import org.apache.axis2.context.MessageContext;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.api.ExceptionCodes;
+import org.wso2.carbon.apimgt.api.gateway.GatewayPolicyDTO;
+import org.wso2.carbon.apimgt.gateway.internal.ServiceReferenceHolder;
+import org.wso2.carbon.apimgt.gateway.service.APIGatewayAdmin;
+import org.wso2.carbon.apimgt.impl.dto.GatewayArtifactSynchronizerProperties;
+import org.wso2.carbon.apimgt.impl.gatewayartifactsynchronizer.ArtifactRetriever;
+import org.wso2.carbon.apimgt.impl.gatewayartifactsynchronizer.exception.ArtifactSynchronizerException;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+
+import java.util.List;
+import java.util.Set;
+
+/**
+ * This class contains the methods used to retrieve artifacts from a storage and deploy and undeploy
+ * the gateway policy mappings in gateway.
+ */
+public class GatewayPolicyDeployer {
+    private static final Log log = LogFactory.getLog(GatewayPolicyDeployer.class);
+    ArtifactRetriever artifactRetriever;
+    private String gatewayPolicyMappingUuid;
+    GatewayArtifactSynchronizerProperties gatewayArtifactSynchronizerProperties;
+
+    public GatewayPolicyDeployer(String gatewayPolicyMappingUuid) {
+        this.artifactRetriever = ServiceReferenceHolder.getInstance().getArtifactRetriever();
+        this.gatewayPolicyMappingUuid = gatewayPolicyMappingUuid;
+        this.gatewayArtifactSynchronizerProperties = ServiceReferenceHolder
+                .getInstance().getAPIManagerConfiguration().getGatewayArtifactSynchronizerProperties();
+    }
+
+    public GatewayPolicyDeployer() {
+        this.artifactRetriever = ServiceReferenceHolder.getInstance().getArtifactRetriever();
+        this.gatewayArtifactSynchronizerProperties = ServiceReferenceHolder
+                .getInstance().getAPIManagerConfiguration().getGatewayArtifactSynchronizerProperties();
+    }
+
+    /**
+     * Deploy gateway policy in the gateway using the deployGatewayPolicy method in gateway admin.
+     *
+     * @throws ArtifactSynchronizerException - If an error occurs while retrieving the gateway policy artifact.
+     * @throws APIManagementException        - If an error occurs while deploying the gateway policy artifact.
```

Comment:
Let's not put a `-` here. Just tab spacing(s) would be fine.

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/12201/commits/d5e8013ecf55f6301164ea05f1537c3b3af25851

Thanks and Regards